src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -123,6 +130,59 @@ static JsonWebKey createJwkFromSettings(Settings settings) throws Exception {
         }
     }
 
+    private static JsonWebKey getSigningJwk(String algorithm, Settings settings) throws IllegalArgumentException {
+        JsonWebKey jwk = new JsonWebKey();
+        jwk.setPublicKeyUse(PublicKeyUse.SIGN);
+
+        if (AlgorithmUtils.isOctet(algorithm)) {
+            jwk.setKeyType(KeyType.OCTET);
+            String signingKey = Optional.ofNullable(settings.get("signing_key")).orElseThrow(() -> new IllegalArgumentException("Settings for signing key is missing. Please specify at least the option signing_key with a shared secret."));
+            jwk.setProperty("k", signingKey);
+            jwk.setAlgorithm(algorithm);
+        } else if (AlgorithmUtils.isEcDsaSign(algorithm)) {
+            jwk.setKeyType(KeyType.EC);
+            jwk.setAlgorithm(algorithm);
+            jwk.setProperty(EC_CURVE, calcCurve(algorithm));
+            jwk.setProperty(EC_PRIVATE_KEY, Optional.ofNullable(settings.get("ec_private")).orElseThrow(() -> new IllegalArgumentException("Settings for EC private key is missing. Please specify ec_private with required x and y coordinates.")));
```

Comment:
For EC: I don't believe we are able to configure default for most values since that would defeat the purpose, but I think we can randomize the private key and starting point. You can correct me if I am wrong.

Reply:
Added option to configure EC using the private key. This way users would need to just add a private key and chosen algorithm.